source/extensions/clusters/dynamic_forward_proxy/cluster.h

```diff
@@ -140,6 +141,38 @@ class Cluster : public Upstream::BaseDynamicClusterImpl,
     const Cluster& cluster_;
   };
 
+  // This acts as the bridge for asynchronous host lookup. If the host is not
+  // present in the DFP cluster, the DFPHostSelectionHandle will receive a onLoadDnsCacheComplete
+  // call unless the LoadDnsCacheEntryHandlePtr is destroyed. Destruction of the
+  // LoadDnsCacheEntryHandlePtr ensures that no callback will occur, at which
+  // point it is safe to delete the DFPHostSelectionHandle.
+  struct DFPHostSelectionHandle
+      : public Upstream::AsyncHostSelectionHandle,
```

Comment:
nit: AsyncHostSelectionHandle is a class, so anything which extends it should also be a class I think.

Reply:
done